backend/app/controllers/reading_lists_controller.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class ReadingListsController < ApplicationController
+  before_action :authorize_request
+
+  def create
+    if reading_list.valid?
+      render(json: { data: reading_list }, status: :created)
+    else
+      render(json: { errors: reading_list.errors.full_messages }, status: :unprocessable_entity)
+    end
+  end
+
+  def update
+    if !user_owns_reading_list?
+      render(json: { errors: ["Unauthorized request detected"] }, status: :unauthorized)
+    elsif reading_list.update(reading_list_params)
+      render(json: { data: reading_list }, status: :ok)
+    else
+      render(json: { errors: reading_list.errors.full_messages }, status: :unprocessable_entity)
+    end
+  end
+
+  def destroy
+    if user_owns_reading_list?
+      reading_list.destroy
+      render(json: { data: { message: "Successfully deleted" } }, status: :ok)
+    else
+      render(json: { errors: ["Unauthorized request detected"] }, status: :unauthorized)
+    end
+  end
+
+  private
+
+  def reading_list_params
+    params.require(:reading_list).permit(:book_id, :status).merge({ user_id: @user.id })
+  end
+
+  def reading_list
+    @reading_list ||= params[:id] ? ReadingList.find_by(id: params[:id]) : ReadingList.create(reading_list_params)
+  end
+
+  def user_owns_reading_list?
+    reading_list.user_id == @user.id
+  end
```

Comment:
Since ReadingList belongs_to user, and User has_many reading_lists, you can take advantage of the ActiveRecord relationship

```
@user.reading_lists.find(params[:id])
```

You can actually make this a method

```
def reading_list
  @reading_list ||= @user.reading_lists.find(params[:id])
end
```

FYI that this will return an error if not found. So make sure you can handle that